protoduneana/verticaldrift/singlehit/MCsingleHitCheater_module.cc

```diff
@@ -0,0 +1,287 @@
+
+    for(int j = 0 ; j < (int) vTrackIdToLabelPair.size() ; j++)
+    {
+      if (vGeneratorLabels[vTrackIdToLabelPair[j].first] == noTrackID )
+      {
+        vGeneratorLabels[vTrackIdToLabelPair[j].first] = vTrackIdToLabelPair[j].second;
+      }
+      else
+      {
+        std::cout << "ISSUE WITH ASSOCIATION " << vTrackIdToLabelPair[j].first << std::endl;
+        vGeneratorLabels[vTrackIdToLabelPair[j].first] = vTrackIdToLabelPair[j].second;
+      }
+
+    }// end for pair(trackID,tag)
+      
+
+    // get vector of generator tag (ie labels)
+    std::vector<std::string> vLabels = HowManyLabels(vTrackIdToLabelPair);
+
+    std::cout << "ASSOCIATION TRACKID GENERATOR TAG MADE ---> HIT ASSOCIATION NOW" <<  std::endl;
+
+    // Hit association
+
+    art::ValidHandle<std::vector<recob::Hit>> HitList_handle = e.getValidHandle<vector<recob::Hit>>(fHitLabel);
+    int fNHits = HitList_handle->size();
+
+    if( LogLevel > 2) std::cout << "THERE ARE " << fNHits << " HITS IN EVENT " << fEventID << std::endl;
+    std::cout << "STARTING GENRATOR HIT ASSOCIATION " << std::endl;
+
+
+    std::vector<int> vNumberOfHits((int) vLabels.size(),0);
+    int NumberOfElecNoiseHit = 0;
+
+    for(int index = 0 ; index < fNHits ; index++)
+    { 
+
+      const recob::Hit& hit = HitList_handle->at(index);
+      std::vector<std::pair<int , float>> tempIDPair;
+
+      for (const auto & ide : bt_serv->HitToEveTrackIDEs(clockData, hit))
+      {
+        //get pair(trckID,energy)
+      	tempIDPair.push_back(std::make_pair(ide.trackID,ide.energy));
+      }
+
+      // sort by energy to have the main MCparticle for each hit
+      std::sort(tempIDPair.begin(), tempIDPair.end(), [](std::pair<int,float> a, std::pair<int,float> b){ return a.second > b.second;});
+
+      if(tempIDPair.size() == 0)
+      {
+        NumberOfElecNoiseHit += 1;
+        continue;
+      }
+      //get tag associated to TrackID 
+      std::string tag = vGeneratorLabels[tempIDPair[0].first];
+
+      for( int k = 0 ; k < (int) vLabels.size() ; k++ )
+      {
+        if( tag == vLabels[k] ) 
+        {
+          vNumberOfHits[k] += 1;
+          break;
+        }
+      }
+    }// end loop on hits
+
+
+    if( LogLevel > 2) std::cout << "THERE ARE " << NumberOfElecNoiseHit << " WITHOUT MCTRUTH  (ie Electronic NOISE)" << std::endl;
+ 
+    int sum = 0;
+    for( int k = 0 ; k < (int) vLabels.size() ; k++ )
+    {  
+      if( LogLevel > 2) std::cout << "THERE ARE " << vNumberOfHits[k] << " HITS ASSOCIATED WITH GENERATOR " << vLabels[k] << std::endl;
+      sum += vNumberOfHits[k];
+    }
+    sum += NumberOfElecNoiseHit;
+    
+    if(sum == fNHits) std::cout << " ALL HITS ARE WELL ASSOCIATED" << std::endl;
+      
+  }// end if e = simu
+}
+
+
+void pdvdana::MCsingleHitCheater::beginJob()
+{
+  // Implementation of optional member function here.
+  art::ServiceHandle<art::TFileService> tfs;
```

Comment:
This might set off Clang when we try to cut a release.  `tfs` looks like an unused variable to me.  Can you remove it?